src/arch/arm/kernel/boot.c

```diff
@@ -214,7 +225,18 @@ BOOT_CODE static bool_t init_cpu(void)
 
 #ifdef CONFIG_ARCH_AARCH64
     /* initialise CPU's exception vector table */
+#if defined(__CHERI_PURE_CAPABILITY__)
+    /* Need to re-build an unsealed exception entry capability */
+    setVtable((pptr_t) cheri_build_code_cap_unbounded((ptraddr_t)arm_vector_table,
+        __CHERI_CAP_PERMISSION_ACCESS_SYSTEM_REGISTERS__ |
+        __CHERI_CAP_PERMISSION_PERMIT_LOAD__ |
+        __ARM_CAP_PERMISSION_MUTABLE_LOAD__ |
+        __ARM_CAP_PERMISSION_EXECUTIVE__ |
+        __CHERI_CAP_PERMISSION_PERMIT_LOAD_CAPABILITY__ |
+        __CHERI_CAP_PERMISSION_PERMIT_EXECUTE__));
+#else
     setVtable((pptr_t)arm_vector_table);
+#endif
```

Comment:
I'd rather introduce a local var of type pptr_t and set that to arm_vector_table, then update it for CHERI and then always call setVtable.